File: sdk-examples/dotnet/AwsEc2Example.cs
// Example: Launching an EC2 instance using AWS SDK for .NET

using Amazon.EC2;
using Amazon.EC2.Model;
using System;
using System.Threading.Tasks;

class AwsEc2Example
{
    static async Task Main()
    {
        var ec2Client = new AmazonEC2Client();

        var launchRequest = new RunInstancesRequest
        {
            ImageId = "ami-0abcdef1234567890", // Replace with a valid AMI ID
            InstanceType = InstanceType.T2Micro,
            MinCount = 1,
            MaxCount = 1
        };

        var launchResponse = await ec2Client.RunInstancesAsync(launchRequest);

        // Print the ID of the launched instance
        foreach (var instance in launchResponse.Reservation.Instances)
        {
            Console.WriteLine($"Launched EC2 instance with ID: {instance.InstanceId}");
        }
    }
}

File: sdk-examples/java/AwsEc2Example.java
// Example: Launching an EC2 instance using AWS SDK for Java v2

import software.amazon.awssdk.services.ec2.Ec2Client;
import software.amazon.awssdk.services.ec2.model.*;

public class AwsEc2Example {
    public static void main(String[] args) {
        Ec2Client ec2 = Ec2Client.create();

        RunInstancesRequest runRequest = RunInstancesRequest.builder()
                .imageId("ami-0abcdef1234567890") // Replace with a valid AMI ID
                .instanceType(InstanceType.T2_MICRO)
                .maxCount(1)
                .minCount(1)
                .build();

        RunInstancesResponse response = ec2.runInstances(runRequest);

        // Print the ID of the launched instance
        String instanceId = response.instances().get(0).instanceId();
        System.out.println("Launched EC2 instance with ID: " + instanceId);

        ec2.close();
    }
}